src/sync-merkleroots.test.ts

```diff
@@ -0,0 +1,315 @@
+import fetchMock from 'jest-fetch-mock';
+import {
+  ErrorStaleLastEvaluatedKey,
+  ErrorSyncMerkleRootsTimeout,
+  ExclusiveStartKeyPage,
+} from '../dist/typescript-npm-package.cjs';
+import { SpvWalletClient } from './client';
+import { MerkleRoot, MerkleRootsRepository } from './types';
+
+const server = 'http://localhost:3003/api/v1';
+
+const client = new SpvWalletClient(server, {
+  xPriv:
+    'xprv9s21ZrQH143K3axKPtYBDKsrAvN3J85z6nZuW5ihYK8JpDWfqHdjswvUnbXzUpMBne1WD6FQmTzymB4Pt3u3UVSauxzq5PswBWr3vYtowmW',
+});
+
+// creates a simulated repository a client passes to SyncMerkleRoots()
+const createRepository = (dataBase: { merkleRoots: MerkleRoot[] }): MerkleRootsRepository => {
+  return {
+    saveMerkleRoots: (syncedMerkleRoots: MerkleRoot[]) => {
+      console.log('\nsaveMerkleRoots called\n');
```

Comment:
*minor*
I don't think that logger is needed for tests - but maybe I'm wrong. 
(The purpose of the tests is to pass - use logger only for development/debugging phase - then ship it)